vllm/config.py

```diff
@@ -2649,11 +2702,52 @@ def __post_init__(self):
                            "Disabling `torch.compile`.")
             self.compilation_config.level = CompilationLevel.NO_COMPILATION
 
+        if self.model_config is not None and \
+            self.compilation_config.level == CompilationLevel.PIECEWISE and \
+            not self.compilation_config.cache_dir:
+            # no provided cache dir, generate one based on the known factors
+            # that affects the compilation. if none of the factors change,
+            # the cache dir will be the same so that we can reuse the compiled
+            # graph.
+            hash_key = self.compute_hash()
+            cache_dir = os.path.join(envs.VLLM_CACHE_ROOT,
+                                     "torch_compile_cache", hash_key)
+            os.makedirs(cache_dir, exist_ok=True)
+            self.compilation_config.cache_dir = cache_dir
+            logger.info("Using cache directory: %s for vLLM's torch.compile",
+                        self.compilation_config.cache_dir)
+
         current_platform.check_and_update_config(self)
 
         if not self.instance_id:
             self.instance_id = random_uuid()[:5]
 
+    def compute_hash(self) -> str:
+        """
+        Provide a hash that uniquely identifies all the configs
+        that affect the model execution, and therefore the compilation.
+        """
+        factors: List[Any] = []
+        # summarize system state
+        from torch._inductor.codecache import CacheBase
+        system_factors = CacheBase.get_system()
+        factors.append(system_factors)
+
+        # summarize pytorch state
+        from torch._inductor.codecache import torch_key
+        torch_factors = torch_key()
+        factors.append(torch_factors)
+
+        # summarize vllm config
```

Comment:
There's a few entries in the SchedulerConfig that could affect the computation graph:
* max_num_batched_tokens
* max_num_seqs
* max_model_len

Not sure if they do currently. But I could see us making decisions that affect the compute graph based on them. 

One thing I have in mind here: When constructing the Flux kernels in https://github.com/vllm-project/vllm/pull/9886 need to supply a maximum `num_tokens`. We could use these for that purpose.

Reply:
It's not in the PR yet but I am using `max_num_batched_tokens` for the flux rewrite.